source/change-streams/change-streams.rst

```diff
@@ -405,28 +454,45 @@ If neither ``startAtOperationTime`` nor ``resumeAfter`` are specified, and the m
 resumeAfter
 ^^^^^^^^^^^
 
-``resumeAfter`` is used to resume a changeStream that has been stopped to ensure that only changes starting with the log entry immediately *after* the provided token will be returned. If the resume token specified does not exist, the server will return an error. 
+``resumeAfter`` is used to resume a change stream that has been stopped to ensure that only changes starting with the log entry immediately *after* the provided token will be returned. If the resume token specified does not exist, the server will return an error.
+
+ChangeStream::resumeToken
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The locally cached ``ChangeStream::resumeToken`` MUST be updated by the following rules.
+
+Upon creating a ``ChangeStream``, if ``resumeAfter`` is set in ``ChangeStreamOptions``, then ``ChangeStream::resumeToken`` MUST be set to the value of ``resumeAfter``.
+
+Upon sending the initial ``aggregate`` or a ``getMore`` command, if the response includes ``resumeTokens``, the following rules apply:
+- If the response has an empty batch (no documents) then ``ChangeStream::resumeToken`` MUST be set to ``resumeTokens[0]`` (the only element in ``resumeTokens``).
+- Otherwise, when document ``i`` of the current batch is returned, ``ChangeStream::resumeToken`` MUST be set to ``resumeTokens[i]`` (the only element in ``resumeTokens``).
```

Comment:
Wouldn't resumeTokens have more than one element in this case?

Reply:
Doh, right. Fixed.